### PR TITLE
Field name

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Anki trim field name when they are added/renamed (but not when they already exists). 

Assuming https://github.com/ankitects/anki/pull/869 gets merged, this PR will get more commits, unless it gets merged quickly.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java [hunk 1]
```diff
@@ -45,6 +45,8 @@
 import com.ichi2.utils.JSONObject;
 
 
+import org.jetbrains.annotations.Nullable;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java:48]: No jetbrains Nullable :-), this is an AndroidStudio trick where it pulls in jetbrains annotation before android annotation, might be a preference somewhere so it never does it

**Review by mikehardy — CHANGES_REQUESTED** [review]
I need to modify how codecov integration adds comments, they are informative in that these lines are not covered and it would be nice to have a test that covered them and checked the new trim case, but they are distracting in review

at any rate, seems fine just the trivial annotation import thing

File: AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java:48]: On the plus side the lint check did fail it so it was noted automatically :-)

```

  /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java:48: Error: Use androidx.annotation.NonNull and androidx.annotation.Nullable. See explanation for IDE-level fix [InconsistentAnnotationUsage]
  import org.jetbrains.annotations.Nullable;
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  
```

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java:48]: Automation is awesome - The Android Studio explanation shows:

https://github.com/ankidroid/Anki-Android/blob/1f785c1b8c324266fdf652e84c9d9309d7278392/lint-rules/src/main/java/com/ichi2/anki/lint/rules/InconsistentAnnotationUsage.java#L37-L38

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java:48]: I do love externalizing memory, that is a fantastic error message haha

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/ModelFieldEditor.java]: Corrected

**Review by mikehardy — APPROVED** [review]

**Pull request merged**